### PR TITLE
feat: User Strategy in Namespace-level ConfigMap

### LINKED ISSUE #177: User Strategy Selection Field
# Summary

Add user strategy selection field to namespace-level ConfigMap.
Add various scenarios to the [design doc](https://docs.google.com/document/d/1BzmSMhqr_Qs4X42TufRwfB8f2RJXaypSXDVxh-VKy3s) (ex: consider different strategies selected for different rollout resources).

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

### PR DESCRIPTION
**Pull request by afugazzotto**
Fixes #177 

### Modifications

Added a namespace-level ConfigMap users can use to add user-related customization parameters. Modified the config watcher to look for ConfigMaps in all namespaces and handle only those of interest based on labels and the namespace they are in.

### Verification

Manually tested.

### CODE REVIEW

File: internal/common/common.go [hunk 1]
```diff
@@ -33,6 +33,9 @@ const (
 	// LabelValueUSDEConfig is the label value used to identify the USDE ConfigMap
 	LabelValueUSDEConfig = "usde-config"
 
+	// LabelValueNamespaceConfig is the label value used to identify the namespace-level ConfigMap
```
Comment (juliev0) [inline internal/common/common.go:36]:
```suggestion
	// LabelValueNamespaceConfig is the label value used to identify the user's namespace-level ConfigMap
```

Comment (juliev0) [reply internal/common/common.go:36]: feel free to re-word, but wanted to contrast from "numaplane-system" namespace for clarity

File: internal/controller/config/config.go [hunk 2]
```diff
@@ -22,6 +22,10 @@ type ConfigManager struct {
 	// USDE Config
 	usdeConfig     USDEConfig
 	usdeConfigLock *sync.RWMutex
+
+	// Namespace-level Config
```
Comment (juliev0) [inline internal/controller/config/config.go:26]:
```suggestion
	// User Namespace-level Config
```

File: internal/util/kubernetes/config_watcher.go [hunk 3]
```diff
@@ -38,7 +40,7 @@ func StartConfigMapWatcher(ctx context.Context, config *rest.Config) error {
 func watchConfigMaps(ctx context.Context, client kubernetes.Interface, namespace string) {
```
Comment (juliev0) [inline internal/util/kubernetes/config_watcher.go:40]: nit: the contract of this function currently makes it seem like we're watching ConfigMaps specifically in the `namespace` that's passed in as a parameter, which we're not. We could either just rename the parameter as `numaplaneNamespace` or something like that, or don't pass in `namespace` at all (the caller is calling `os.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")` which could theoretically happen here)

File: internal/usde/usde.go [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package usde
```
Comment (afugazzotto) [inline internal/usde/usde.go:1]: This file will later also contain the USDE logic and more USDE related code.

**Review by juliev0 — APPROVED** [review]

**Review by xdevxy — APPROVED** [review]

**Pull request merged**